### DISCUSSION TITLE
New homebrew install on Macbook air M1: various problems

### DISCUSSION THREAD

**Discussion by jalvarez13** [Homebrew/discussions#4140] [Getting started] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.20
ORIGIN: https://github.com/Homebrew/brew
HEAD: 344d32bf7f40183359592ef81ca6486e0587981b
Last commit: 8 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: b22a73f1d83baafd6d338b0e16999aba988645f9
Core tap last commit: 3 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.85.0 => /usr/bin/curl
macOS: 13.1-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

After firing up the Mac Terminal, I have to switch to my admin account (from which I installed Homebrew) and execute 'source ~/. zprofile' . Oherwise the brew command produces a 'zsh: command not found: brew' response.

Also, when doing an emacs cask install, even though the installation finishes without any warnings, I can't run emacs (see my post about it at https://github.com/Homebrew/discussions/discussions/4063)

Finally, after installing the regular emacs, even though I can run it from my admin account, when I try to use it from my normal user account, it fails to find it.

**Comment by SMillerDev** [comment]
> After firing up the Mac Terminal, I have to switch to my admin account (from which I installed Homebrew) and execute 'source ~/. zprofile' . Oherwise the brew command produces a 'zsh: command not found: brew' response.

Then you skipped the instructions at the end of the install.

**Reply by jalvarez13** [reply]
Hello Sean. No, I didn't skip them. I executed all three of them.

**Comment by gromgit** [comment, marked as answer]
> After firing up the Mac Terminal, I have to switch to my admin account (from which I installed Homebrew) and execute 'source ~/. zprofile' . Oherwise the brew command produces a 'zsh: command not found: brew' response.

`.zprofile` is only sourced for a login shell, so you'd have to logout and login again with the admin account, for it to be automaticallly sourced. Alternatively, switch to the admin user on the command line via `su -l <admin_user>`.

> Finally, after installing the regular emacs, even though I can run it from my admin account, when I try to use it from my normal user account, it fails to find it.

That's because your `PATH` is not set correctly. The _Next steps_ instructions that the Homebrew installer printed will only take effect for the (admin) user you ran the installer with; the `~` in `~/.zprofile` means "the home directory of the user currently logged in". For all other users who want to reap the benefits of Homebrew-installed software, see the `brew shellenv` command line that was added to your admin user's `.zprofile`, then add the same line to each user's `.bash_profile` or `.profile` (if the user is running `bash`) or `.zprofile` (if running `zsh`).

**Comment by jalvarez13** [comment]
Thank yoiu Adrian. Your suggestions worked perfecly!  

I'm stll struggling with the emacs cask install though (https://github.com/orgs/Homebrew/discussions/4063)

**Comment by brimobile1** [comment]
> ### Output of `brew config`
> 
> ```shell
> HOMEBREW_VERSION: 3.6.20
> ORIGIN: https://github.com/Homebrew/brew
> HEAD: 344d32bf7f40183359592ef81ca6486e0587981b
> Last commit: 8 days ago
> Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
> Core tap HEAD: b22a73f1d83baafd6d338b0e16999aba988645f9
> Core tap last commit: 3 hours ago
> Core tap branch: master
> HOMEBREW_PREFIX: /opt/homebrew
> HOMEBREW_CASK_OPTS: []
> HOMEBREW_MAKE_JOBS: 8
> Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
> CPU: octa-core 64-bit arm_firestorm_icestorm
> Clang: 14.0.0 build 1400
> Git: 2.37.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
> Curl: 7.85.0 => /usr/bin/curl
> macOS: 13.1-arm64
> CLT: 14.2.0.0.1.1668646533
> Xcode: 14.2
> Rosetta 2: false
> ```
> 
> 
> ### Output of `brew doctor`
> 
> ```shell
> Your system is ready to brew.
> ```
> 
> 
> ### Description of issue
> 
> After firing up the Mac Terminal, I have to switch to my admin account (from which I installed Homebrew) and execute 'source ~/. zprofile' . Oherwise the brew command produces a 'zsh: command not found: brew' response.
> 
> Also, when doing an emacs cask install, even though the installation finishes without any warnings, I can't run emacs (see my post about it at https://github.com/Homebrew/discussions/discussions/4063)
> 
> Finally, after installing the regular emacs, even though I can run it from my admin account, when I try to use it from my normal user account, it fails to find it.
